scripts/build/Jenkins/Jenkinsfile

```diff
@@ -336,51 +416,281 @@ def PreBuildCommonSteps(Map pipelineConfig, String snapshot, String repositoryNa
     // we need to cleanup. Once all branches take this relocation, we can remove this
     if(env.CLEAN_WORKSPACE?.toBoolean() || fileExists("${workspace}/.git")) {
         if(fileExists(workspace)) {
-            palRmDir(workspace)
+            PlatformRmDir(workspace)
         }
     }
 
     dir(workspace) {
+
         // Add folder where we will store the 3rdParty downloads and packages
-        if(!fileExists('3rdParty')) {
-            palMkdir('3rdParty')
+        dir('3rdParty') {
         }
-    }
-    dir("${workspace}/${ENGINE_REPOSITORY_NAME}") {
-        CheckoutRepo(disableSubmodules)
 
-        // Get python
-        if(env.IS_UNIX) {
-            sh label: 'Getting python',
-               script: 'python/get_python.sh'
-        } else {
-            bat label: 'Getting python',
-                script: 'python/get_python.bat'
+        //get the engine repo first and get python, set the env.CHANGE_ID and env.CHANGE_DATE only if the repo is the engine repo
+        dir(ENGINE_REPOSITORY_NAME) {
+            echo "Checkout Engine: ${ENGINE_URL}"
+            def setEnvCommit = false
+            def the_branch = branchName
+            if(ENGINE_COMMIT) {
+                //the pr is for the engine repo, so we want to use the passed in branch name and record the change id and change date
+                setEnvCommit = true
+            } else {
+                //this pr is for another repo so we want to use the development branch for the engine and do not set the commit id or change date
+                the_branch = ENGINE_DEVELOPMENT_BRANCH
+            }
+
+            UnInstallGitLFS() // Prevent git from pulling lfs objects during checkout
+
+            //checkout the engine branch
+            CheckoutRepo(the_branch, GetEngineRemoteConfig(), disableSubmodules, setEnvCommit)
+
+            // Get python and clean.
+            // Always run the clean step, the scripts detect what variables were set, but it also cleans if
+            // the NODE_LABEL has changed
+            if(env.IS_UNIX) {
+                sh label: 'Getting python',
+                   script: "${pipelineConfig.PYTHON_DIR}/get_python.sh"
+                sh label: "Running ${platform} clean",
+                   script: "${pipelineConfig.PYTHON_DIR}/python.sh -u ${pipelineConfig.BUILD_ENTRY_POINT} --platform ${platform} --type clean"
+            } else {
+                bat label: 'Getting python',
+                    script: "${pipelineConfig.PYTHON_DIR}/get_python.bat"
+                bat label: "Running ${platform} clean",
+                    script: "${pipelineConfig.PYTHON_DIR}/python.cmd -u ${pipelineConfig.BUILD_ENTRY_POINT} --platform ${platform} --type clean"
+            }
+
+            InstallAndPullGitLFS()
         }
+        dir(EXTRAS_REPOSITORY_NAME) {
```

Comment:
This PR will introduce a significant change to the AR pipeline, specifically with it depending on another repo for it to run. I recommend creating a proposal for this multi-repo setup so that we can discuss it with sig-build. 

When setting up the AR we wanted to avoid having it depend on external repos that can introduce build/test failures and then block merges to the O3DE repo. For example, the project repos run their own pipelines and cannot block any PRs in the O3DE repo from failing the AR. 

We can setup something similar for o3de-extras so that PRs created in that repo run on their own AR pipeline. Failures introduced in the dev branch here would not block the O3DE AR pipeline. A compromise could involve creating a release in o3de-extras that downstream repos would consume. 

If the goal is to actually run this on a separate pipeline, then we can avoid adding the o3de-extras configs in this repo. The o3de-extras repo can have its own Jenkinsfile. There may be some config duplication, but sig-build is working on a setup so other repos can pull the main pipeline template from a central location.

Reply:
This will introduce a significant change to the AR pipeline and it will have a secondary repo dependency, however the philosophy here is that they are treated as the same repo. More like a master repo and a submodule. A gem in the extras is as though it is in the engine. The only difference is what is core engine versus optional. They all still have to work. Extras will be part of the release going forward. I think it defeats the purpose of AR if I can check in a change that will break the extras. The facility for adding further repos such as a project repo can all be done via the .automatedtesting.json in the root of your repo. This way we do not have to maintain multiple jenkinsfiles, and neither do our customers. Since Jenkins has a plugin that allows it, I would also recommend we remove the jenkinsfiles from all the repos.

Reply:
Discussed this in-depth with Colin and sig-build on discord. I think we were able to address my concerns with external repo dependencies, mainly its potential to impact developer velocity. We agreed to move forward with this setup with the following assumptions:

- There are certain gems in the o3de-extras repo that we want to test with each PR to O3DE
- Not all gems added to the o3de-extras repo will be required to pass to merge PRs to O3DE
- The automatedtesting.json file will be used to determine which gems will be tested in the O3DE AR
- There is a low risk of having a change that requires simultaneous changes to both repos that would be blocked if submitted individually. 

I'll take another look at the PR to see if we need to make any other adjustments prior to merging this in.

Reply:
Some additional points from this discussion:
- EBS Snapshots from o3de will still be utilized for the o3de-extras builds. The build artifacts are the output from both o3de and o3de-extras, so there will be overlap in terms of reusing artifacts
- SIG-Build will help to get the o3de-extras AR working, but this shouldn't directly impact this PR. We do want this to be working when gems are to be tested, but it should not block the current PR
- The o3de PR is currently passing in its branch, but this will go through AR to validate, of course